### PR TITLE
[Feature][Plugin] Add plugin-spark-source-redis

### PR DESCRIPTION
**Pull request by wuchunfu**
---

## Purpose of this pull request

This pull request add plugin-spark-source-redis. related: #751

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] Change does not need document change, or I will submit document change to https://github.com/InterestingLab/seatunnel-docs later

### CODE REVIEW

**Comment by wuchunfu** [comment]
cc @RickyHuo @CalvinKirs

File: plugin-spark-source-redis/src/main/scala/io/github/interestinglab/waterdrop/spark/source/Redis.scala [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import com.redislabs.provider.redis.{RedisConfig, RedisEndpoint, toRedisContext}
+import io.github.interestinglab.waterdrop.common.config.CheckResult
+import io.github.interestinglab.waterdrop.spark.SparkEnvironment
+import io.github.interestinglab.waterdrop.spark.batch.SparkBatchSource
+import org.apache.spark.sql.{Dataset, Row}
+
+class Redis extends SparkBatchSource {
+  val defaultPort: Int = 6379
+  val defaultDb: Int = 0
+  val defaultPartition: Int = 3
```
Comment (CalvinKirs) [inline plugin-spark-source-redis/src/main/scala/io/github/interestinglab/waterdrop/spark/source/Redis.scala:29]: It is best to provide the redis.timeout parameter

Comment (wuchunfu) [reply plugin-spark-source-redis/src/main/scala/io/github/interestinglab/waterdrop/spark/source/Redis.scala:29]: OK, let me add it

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**